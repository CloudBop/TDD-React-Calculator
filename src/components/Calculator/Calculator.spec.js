import React from 'react';
import { shallow } from 'enzyme';
import Calculator from './Calculator';
import Display from '../Display/Display'
describe('Calculator', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Calculator />));
  it('Should compare against snapshot <Calculator/>', () => expect(wrapper).toMatchSnapshot())
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it("should render the Display Component", ()=> {
    // access the state variables and methods of Calculator by utilizing the instance method on the wrapper object.
    expect(wrapper.containsMatchingElement(<Display displayValue={wrapper.instance().state.displayValue}/>)).toEqual(true)
  })
});