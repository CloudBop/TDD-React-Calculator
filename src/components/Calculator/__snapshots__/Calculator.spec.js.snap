// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Should compare against snapshot <Calculator/> 1`] = `ShallowWrapper {}`;
